sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -148,7 +150,7 @@ object RewritePredicateSubquery extends Rule[LogicalPlan] with PredicateHelper {
           // will have the final conditions in the LEFT ANTI as
           // (A.A1 = B.B1 OR ISNULL(A.A1 = B.B1)) AND (B.B2 = A.A2) AND B.B3 > 1
           val finalJoinCond = (nullAwareJoinConds ++ conditions).reduceLeft(And)
-          Join(outerPlan, newSub, LeftAnti, Option(finalJoinCond), JoinHint.NONE)
+          Join(outerPlan, newSub, LeftAnti, Option(finalJoinCond), JoinHint(None, subHint))
```

Comment:
Shall we add comment here as well?
> Add subquery hint as right hint as the subquery plan is on the right side of the join

Maybe add a method `def createJoinHint(subHint ...)`, so that we don't need to duplicate the comment in many places: ``

Reply:
Not sure how useful this is, as from the code it is more or less apparent where the subquery hint goes; if the subquery goes on the right side, then the hint goes on the right side too.